spec/unit/plugins/darwin/hardware_spec.rb

```diff
@@ -0,0 +1,115 @@
+#
+# Author:: Nate Walck (<nate.walck@gmail.com>)
+# Copyright:: Copyright (c) 2016-present Facebook, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require File.expand_path(File.dirname(__FILE__) + "/../../../spec_helper.rb")
+require File.expand_path("#{File.dirname(__FILE__)}/hardware_system_profiler_output.rb")
+
+describe Ohai::System, "Darwin hardware plugin", :unix_only do
+  let (:plugin) { get_plugin("darwin/hardware") }
+  before(:each) do
+    # Make sure it always runs correct commands and mock the data as it calls them
+    allow(plugin).to receive(:shell_out).with(
+      "system_profiler SPHardwareDataType -xml"
+    ).and_return(
+      mock_shell_out(0, HardwareSystemProfilerOutput::Hardware, "")
+    )
+
+    allow(plugin).to receive(:shell_out).with(
+      "sw_vers -productName"
+    ).and_return(
+      mock_shell_out(0, "Mac OS X", "")
+    )
+
+    allow(plugin).to receive(:shell_out).with(
+      "sw_vers -productVersion"
+    ).and_return(
+      mock_shell_out(0, "10.12", "")
+    )
+
+    allow(plugin).to receive(:shell_out).with(
+      "sw_vers -buildVersion"
+    ).and_return(
+      mock_shell_out(0, "16A239j", "")
+    )
+
+    allow(plugin).to receive(:shell_out).with(
+      "uname -m"
+    ).and_return(
+      mock_shell_out(0, "x86_64", "")
+    )
+
+    allow(plugin).to receive(:shell_out).with(
+      "system_profiler SPStorageDataType -xml"
+    ).and_return(
+      mock_shell_out(0, HardwareSystemProfilerOutput::Storage, "")
+    )
+
+    allow(plugin).to receive(:shell_out).with(
+      "system_profiler SPPowerDataType -xml"
+    ).and_return(
+      mock_shell_out(0, HardwareSystemProfilerOutput::Power, "")
+    )
+  end
+
+  it "should parse hardware data correctly" do
```

Comment:
(Minor style comment) In alignment with [BetterSpecs](http://betterspecs.org/#should) we prefer to not use "should" in test descriptions.